protos/core.proto

```diff
@@ -424,4 +429,12 @@ message TableMetadata {
   Target target = 1;
   string type = 2; // "table" or "view"
   repeated Field fields = 3;
+  int64 lastUpdatedMillis = 4;
+}
+
+message PersistedTableMetadata {
+  TableMetadata table = 1;
+  string create_query = 2;
```

Comment:
I've just realized a problem.
Storing just the query is not enough, you need to also check the pre_operations and post_operations, as well as other metadata such as the table type, any warehouse specific settings (partitionBy, clusterBy).

These are also going to vary between each table type (view, incremental, table, as well as declarations, assertions and operations).

Therefore, I'd strongly suggest storing a hash instead. Storing all of these things is going to be a pain in the ass, whereas computing a hash from a JSON object is much easier...

If you still want to store the exact query, then you could possibly store the list of queries in the "tasks" object of an ExecutionAction

Reply:
I changed the name to create_query_hash, which will store the hash of the publish task object.